File: IPL_WEEK_5/E/main.cpp
#include<iostream>
#include<string>
#include<vector>

using namespace std;

int main() {
	int n;
	cin >> n;
	vector<string> names;
	for(int i = 0; i < n; i++){
		string tmp;
		cin>>tmp;
		cout<<tmp<<endl;
		names.push_back(tmp);
		//names.push_back(tmp);
	}
	cout<<names.size();
	/*int min = 10000;
	for(int i = 0; i < names.size(); i++) {
		int count = 0;
		cout<<names[i];
		for(int j = 0; j < names.size(); j++){
			if(i==j){
				continue;
			}
			string tmp = names[j];
			while(names[i].compare(tmp)!=0){
				char c = tmp[0];
				tmp.erase(0);
				tmp.push_back(c);
				count++;
			}
			cout<<count<<endl;
		}
		if(count < min){
			min = count;
		}
	}
	cout<<min<<endl;*/
	return 0;
}

File: IPL_WEEK_5/E/Main.java
import java.io.*;
import java.util.*;

public class Main {
  public static void main(String[] args) {
    Scanner in = new Scanner(System.in);
    int n = in.nextInt();
    String[] names = new String[n];
    for(int i = 0; i < names.length; i++) {
      names[i] = in.next();
      //System.out.println(names[i]);
    }
    int min = Integer.MAX_VALUE;
    for(int i = 0; i < names.length; i++) {
      int count = 0;
      for(int j = 0; j < names.length; j++) {
        if(i==j){
          continue;
        }
        String tmp = names[j];
        for(int k = 0; k < tmp.length(); k++){
          if(tmp.equals(names[i])){
            break;
          }
          char c = tmp.charAt(0);
          tmp = tmp.substring(1,tmp.length())+c;
          count++;
        }
        if(!tmp.equals(names[i])) {
          System.out.println(-1);
          return;
        }
      }
      if(count < min) {
        min = count;
      }
    }
    System.out.println(min);
  }
}
